lib/main.coffee

```diff
@@ -1,114 +1,80 @@
-                    Promise.all(
-                      entries.filter (entry) -> entry.isFile() and path.extname(entry.getBaseName()) is '.cson'
-                        .map (entry) -> promiseHelper.readCsonFile(entry.path)
-                    )
-                  .then (grammars) ->
-                    # Remove grammars
-                    grammars.map (grammar) ->
-                      {scopeName} = grammar
-                      atom.grammars.removeGrammarForScopeName scopeName
-                  .then ->
-                    # Remove loaded package (Hack force reload)
-                    delete atom.packages.loadedPackages[grammarsPackageName]
-
-                    # Load package
-                    # (use `loadGrammarsSync` instead of `loadGrammars` because `loadGrammars` doesn't work properly)
-                    atom.packages.loadPackage(grammarsPackageName).loadGrammarsSync()
-                  .then =>
-                    # Reload grammars for each editor
-                    atom.workspace.getTextEditors().forEach (editor) =>
-                      if editor.getGrammar().packageName is grammarsPackageName
-                        if @debug then console.log editor.getTitle()
-                        atomVersion = parseFloat(atom.getVersion())
-                        if atomVersion < 1.11
-                          editor.reloadGrammar()
-                        else
-                          # Workaround because:
-                          # - `reloadGrammar` is buggy before 1.11 (https://github.com/atom/atom/issues/13022)
-                          # - `maintainGrammar` change this behavior and don't reload existing grammar.
-                          #   - https://github.com/atom/atom/pull/12125
-                          #   - https://github.com/atom/atom/blob/c844d0f099e6ed95c52f0b94e1f141759926aeb8/src/text-editor-registry.js#L201
-                          grammarOverride = atom.textEditors.editorGrammarOverrides[editor.id]
-                          atom.textEditors.setGrammarOverride editor, 'text.plain'
-                          if grammarOverride?
-                            atom.textEditors.setGrammarOverride editor, grammarOverride
-                          else
-                            atom.textEditors.clearGrammarOverride editor
-
-                    Promise.resolve 'success'
-              else
-                Promise.resolve projectPackage.name + ' is not the right package.'
-        .catch (error) =>
-          if @debug then console.error error
-          Promise.resolve packageJsonPath + " doesn't exists."
-
-    Promise.all(promises)
-      .then (msg) =>
-        console.log 'Grammars reloaded.'
-        if @debug then console.log msg
-      .catch (error) ->
-        console.log 'Grammars failed to reload.'
-        console.error error
-
-    deactivate: ->
-      @subscriptions?.dispose()
-      @liveReloadSubscriptions?.dispose()
+    return unless atom.inDevMode()
```

Comment:
Could you explain why you removed  `and not atom.inSpecMode()`

Reply:
No I can't. Can you explain what that did? I don't know what "spec mode" is or why it should be avoided.

Reply:
Spec mode it's the mode where the tests are run.

Could you restore this condition.

Reply:
Okay, that's what I suspected. But why shouldn't the package be enabled during tests? Perhaps tests should be added that actually test the core functionality.

Reply:
If you want add tests on the core functionality, it would make me happy, but it's not very simple

Reply:
I don't want the package be enable during the tests of others packages.